projects/plugins/jetpack/extensions/shared/memberships/settings.js

```diff
@@ -165,52 +177,61 @@ function NewsletterAccessSetupNudge( { stripeConnectUrl, isStripeConnected, hasN
 	}
 }
 
-function NewsletterAccessRadioButtons( {
+export function NewsletterAccessRadioButtons( {
 	onChange,
 	accessLevel,
 	hasNewsletterPlans,
 	stripeConnectUrl,
 	isEditorPanel = false,
+	hasPaywallBlock = false,
+	paywallBlockSettings = false,
 } ) {
 	const isStripeConnected = stripeConnectUrl === null;
 	const instanceId = useInstanceId( NewsletterAccessRadioButtons );
 
 	return (
 		<fieldset className="editor-post-visibility__fieldset">
 			<VisuallyHidden as="legend">{ __( 'Audience', 'jetpack' ) } </VisuallyHidden>
-			{ Object.keys( accessOptions ).map( key => (
-				<div className="editor-post-visibility__choice" key={ key }>
-					<input
-						value={ key }
-						type="radio"
-						checked={ key === accessLevel }
-						className="editor-post-visibility__radio"
-						id={ `editor-post-${ key }-${ instanceId }` }
-						name={ `editor-newsletter-access__setting-${ instanceId }` }
-						aria-describedby={ `editor-post-${ key }-${ instanceId }-description` }
-						disabled={
-							key === accessOptions.paid_subscribers.key &&
-							( ! isStripeConnected || ! hasNewsletterPlans )
-						}
-						onChange={ event => {
-							const obj = {};
-							obj[ META_NAME_FOR_POST_LEVEL_ACCESS_SETTINGS ] = event?.target?.value;
-							return onChange && onChange( obj );
-						} }
-					/>
-					<label
-						htmlFor={ `editor-post-${ key }-${ instanceId }` }
-						className="editor-post-visibility__label"
-					>
-						{ accessOptions[ key ].label }
-					</label>
-					{ key === accessLevel && key !== accessOptions.everybody.key && (
-						<p className="editor-post-visibility__notice">
-							<NewsletterNotice accessLevel={ accessLevel } />
-						</p>
-					) }
-				</div>
-			) ) }
+			{ Object.keys( accessOptions ).map(
+				key =>
+					( ! paywallBlockSettings || key !== accessOptions.everybody.key ) && (
```

Comment:
Seems this should work but wondering if it makes sense for the `! paywallBlockSettings` check to be outside of the map() function? Seems like if paywallBlockSettings === false, we don't render anything from this map function.

Reply:
Answering my own question: no because we want to use it combination with properties for each accessOption. It's how we don't show everyone only when  loading paywall block.